packages/wrangler/e2e/startWorker.test.ts

```diff
@@ -164,34 +180,101 @@ describe.each(OPTIONS)("DevEnv", ({ remote }) => {
 
 		const inspectorUrl = await worker.inspectorUrl;
 
-		let ws = new WebSocket(
-			`ws://${inspectorUrl.host}/core:user:${worker.config.name}`,
-			{ setHost: false, headers: { Host: "example.com" } }
-		);
+		let ws = new WebSocket(inspectorUrl.href, {
+			setHost: false,
+			headers: { Host: "example.com" },
+		});
 
 		let openPromise = events.once(ws, "open");
 		await expect(openPromise).rejects.toThrow("Unexpected server response");
 
 		// Check validates `Origin` header
-		ws = new WebSocket(
-			`ws://${inspectorUrl.host}/core:user:${worker.config.name}`,
-			{ origin: "https://example.com" }
-		);
+		ws = new WebSocket(inspectorUrl.href, { origin: "https://example.com" });
 		openPromise = events.once(ws, "open");
 		await expect(openPromise).rejects.toThrow("Unexpected server response");
 		ws.close();
 	});
+
+	// Regression test for https://github.com/cloudflare/workers-sdk/issues/5297
+	// The runtime inspector can send messages larger than 1MB limit websocket message permitted by UserWorkers.
+	// In the real-world, this is encountered when debugging large source files (source maps)
+	// or inspecting a variable that serializes to a large string.
+	// Connecting devtools directly to the inspector would work fine, but we proxy the inspector messages
+	// through a worker (InspectorProxyWorker) which hits the limit (without the fix, compatibilityFlags:["increase_websocket_message_size"])
+	// By logging a large string we can verify that the inspector messages are being proxied successfully.
+	it("InspectorProxyWorker can proxy messages > 1MB", async (t) => {
+		t.onTestFinished(() => worker?.dispose());
+
+		const originalConsoleLog = console.log;
+		const mockConsoleLogImpl = (message: unknown, ...args: unknown[]) => {
+			if (typeof message === "string" && /z+/.test(message)) {
+				return; // don't log chunks of the large string
+			}
+
+			originalConsoleLog(message, ...args);
+		};
+		vi.spyOn(console, "info").mockImplementation(mockConsoleLogImpl);
+		vi.spyOn(console, "log").mockImplementation(mockConsoleLogImpl);
+
+		const LARGE_STRING = "This is a large string" + "z".repeat(2 ** 20);
```

Comment:
Even though the limit is explained as "1MB" it is actually 2 ** 20. I was getting unreliable results with `1_000_000` (probably because the rest of the json message _sometimes_ pushed it over the threshold) – it wasn't enough to ensure the message size was over the limit 100% of the time. I will add a comment in a follow-up PR.

Reply:
Oh got it, np 👍